<a href="https://colab.research.google.com/github/oluseyedev/oluseyedev/blob/main/completed_Python_Basics_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Python Basics 1

Author: Dr. Pradeep Hewage

Contact: P.Hewage@bolton.ac.uk

Institution: University of Bolton

## Introduction

This notebook has been written to support Python basics for Data Science module. As we work through the module, we'll often use Google Colab resources to accelerate learning.
<br/>

Programming is a practical activity. It's best to learn by doing, and by running examples at your own pace, in your own time.
<br/>

## What is Google Colab?
Google Colab provides a software environment you can use to execute code. This means you don't have to setup any complicated software environments for yourself - you can simply load this site and run our activities. You'll need your own Google ID to login and use this resource to its full potential. So please, sign up for a Google account if you **do not** already have one.
<br/>

# Using This Resource

I advise that you step through each cell in this notebook slowly, at your own pace. Once you understand each cell, move on. This is important as each cell builds upon the next. Industry activities are provided near the end of the notebook. Enjoy!
<br/>

Remember, each cell is supposed to be executed in turn from the top to the bottom of this notebook. So, keep that in mind!

## 0. The basics

Each cell in the Colab can contain an executable piece of Python code. When we execute a cell, the results of that execute are remembered. Consider the example below. Read it, then execute the cell.

In [ ]:
x = 0
y = 0

z = x + y

print(z)

This is some pretty straightforward code that sets $z$ equal to the value of $x+y$. Since both $x$ and $y$ are defined to be zero, then $z=0$. Suppose we then write the following code. Read it, then execute the cell. What output do you expect?

In [ ]:
x = 1

print(z)

Does it surprise you that $z$ is still zero? It shouldn't - just because we set the value of $x$ to 1, doesn't mean that change is propagated to $z$. We have to explicitly update $z$ to achieve that. Let's try to do that in the cell that follows. Read it, then execute it when ready.

In [ ]:
z = x + y

print(z)

You can see here that this cell has "remembered" that we set $x=1$ in a previous cell. That's because the code we write in cells, and the assignments they define, are stored in memory. Look up at the top right of the browser screen - can you see the Colab telling you your Random Access Memory (RAM) usage? It also shows your disk usage too! Ok, back to this example. What should be clear from this example, is that the order in which we execute cells matters. It is also important that we define all of the items we refer to in cells. For example, try to run the cell below. It will report an error - do you understand why?

In [ ]:
z = x + y - a
print(z)

We get a “name error” here, because we use a term $a$ which we've not introduced before, nor defined. Computers do not run on magic. They only know the things that we tell them. Thus, since I didn't say what a was, or what it equals, the calculation could not be completed.

<br>

Get use to error messages such as this. Encountering them is an important part of learning to program. The question is, how will you respond error messages - panic? Try to react constructively – ask “what is it I don't understand”, “why did this happen”, “what does it mean - what is the computer trying to tell me”. In this case it was clear, "a is not defined". Sometimes error messages are less clear, so be prepared for that. Confusion is entirely normal. I wasn't born knowing this stuff :)


## 1. Comments

Comments are used to help describe the code we write. Why do we needs comments in our code? There are a variety of important reasons.
* To explain how it works.
* To make it easier to read.
* To help make the code easier to maintain, for those who didn’t write it.
* It is important to write informative, succinct comments.
* There are two types of comment in Python:
  * In-line comments
  * And Block comments

<br/>

The hash (#) symbol is used to start a comment.

<br/>

An **in-line comment** explains a single line of code. It may explain the purpose of the line or provide important information and pointers. Such comments should be used only when required.

<br/>

A **block comment** explains one or more lines of code. Blocks can spread over multiple lines if required, depending on the complexity of the code. If a comment needs more than one paragraph, split the comment up using an empty comment line.


Follow the cells below and execute them as you go. The code cells will contain comments that explain how/why things are done. Looking through the cells for yourself will get you into the habit of reading code early on.


In [ ]:
# This is a single line comment.

# Comment that explains the code below, were we simply add two numbers.
number_1 = 10
number_2 = 15
result = number_1 + number_2

# Now we have a multiline comment. I often use these to explain code that
# may not be clear. Ideally, we keep such comments succinct. However, I'll
# break that rule for educational purposes in this notebook.
#
# Below is some code that checks if a character is in a string. If it is
# present, it returns the index corresponding to the position in the string
# where the character was found. For example, if we have the following String:
#
# "ABC"
#
# Then we can view the strings as a collection of characters with specific
# positions, or indexes, in the string, e.g.
#
# index:   0  1  2
#          |  |  |
#          v  v  v
#        " A  B  C "
#
# So, if asking for the index of the character "R" in the string, the index
# will be zero.
text = "ABC"
print("Position of A: ", text.find('A'))
print("Position of B: ", text.find('B'))
print("Position of C: ", text.find('C'))


Can you comment the code below? What would you say? The answer is provided in a few cells down - but don't look right away! Don't worry about not understanding everything the code does - just read what is says, even read it outload in English. Maybe running the code and seeing the output, will help you understand how its written.

In [ ]:
number_1 = 0
number_2 = 10

while number_1 < number_2:
  number_1 = number_1 + 1

  if number_1 % 2 == 0:

    print("number_1 now even:", number_1)

My comments are below. Different people will write comments in different ways. That's ok, so long as the key pieces of information are conveyed.

In [ ]:
# Some simple code that prints out the even numbers between
# number_1 and number_2. The values of these can be altered
# as required.
number_1 = -1
number_2 = 10

# While number_1 is less than number_2
while number_1 < number_2:
  number_1 = number_1 + 1 # increment the value of number_1 by 1.

  # If number_1 is even, print that out.
  if number_1 % 2 == 0:

    print("number_1 now even:", number_1)

Remember, there are single line and multi-line comments in Python. Here's an example of both.

In [ ]:
# This is a single line comment in Python.


'''
This is also a multi-line comment in python.
But we use these sorts of comments to document
specific parts of the code. We'll address this later.
'''

## Activity 1

Here's a simple first activity to get you going. Write a piece of code that computes the average of the numbers 1, 2, 3, 4, 5. Print out the average and comment your code. I Python, we can use the $/$ symbol to represent division. I provide an example answer below.

In [ ]:
# Your awesome code here.

My answer...

In [ ]:
# The average, or mean, is simply the sum of the numbers to
# be averaged, divided by the total number of numbers.
# Simple average calculation
avg = (1 + 2 + 3 + 4 + 5) / 5
print(avg)

## 2. Lines of Code
In Python, lines of code terminate with the end of the line. Any comments that appear in a line are ignored by the interpreter when reading code, so these don't interfere with our line endings. When writing code in the Colab code cells, whenever we press the enter key, we effectively create a new line. Let's consider some examples to support this intuition. Read the line of code below - what do you think will be printed when we run it?

In [1]:
# Output age
age = 34 # age = 35
print(age)

34


Notice that when we run this code, the value 34 was printed out. Thus, the code I wrote after the hash symbol (#) was ignored by the interpreter since it signified a comment line. This is why the value of age wasn't updated. This example contained two distinct lines of code.

Sometimes when writing code, for human readability reasons we may want to break code up over one or more lines. But we want the interpreter to treat the code as a single line. It is possible to this in Python with the line continuation character. This is represented using a backslash ($$). When the interpreter reads a continuation character, it realises that we want it to continue reading any code that follows as belonging to the same line. Take a look at the code below and then execute it. Is the output as you expect?

In [ ]:
x = 4 + \
5
print(x)

Using this continuation line, we are able to split up a line for readability, but have it treated as a single line by the interpreter. For such simple examples this is a bit of daft thing to do. We don't need the backslash here - but in real world examples, we often need to use this. What happens if we don't use a continuation character here? Run the code below to find out.

In [ ]:
x = 4 +
5
print(x)

Because we didn't use the continuation character, we encountered a syntax error. This is the interpreter’s way of telling us the syntax of our code is invalid. It's clear why - even in regular mathematics, a statement such as 4 + is not valid unless there is another term following the plus sign. Whilst the following line that just contains the number 5 is also invalid. It's a little like writing a sentence in English like:

the

It has no meaning and syntactically invalid (no full stop etc). We most often use continuation characters when printing out long lines of text. For instance, if providing the user of an application with guidance or key information. An example of this is provided below. Run the code to double check the output.

In [ ]:
print("Sometimes we may have a great deal of text that we wish to print out "\
      "but according to Python style guides, we shouldn't write lines of code "\
      "greater than 79 characters long. Hence using the continuation character, "\
      " I can get around this restriction.")

Notice how all this information is printed out on one line?

Also notice how I had to break up the lines too? When we print out text using the, print() command, the text must be placed between double quotes, "like this". This allows the interpreter to figure out that anything between the double quote pairs, is to be treated as text. For instance, 5 is a number but "5" is text containing the character 5. I can do maths with the number 5, but I can't do maths with the text "5" (we'll try that shortly!).

We accept that the text I wanted to print has to be within a of double quotes. When the interpreter sees the quote character (") it expects the following information it reads to be text until it sees another double quote character.

I couldn't therefore put the continuation character in-between the double quotes. If I had then the interpreter wouldn't have seen the continuation character as a request to carry on reading the next line, but as a backslash to include in some text.

The output of my code here is also a bit difficult to read, since the output line is very long. I can fix that using the newline character. This is specified using \n. Take a look at the following code sample and then run it.

In [ ]:
print("Sometimes we may have a great deal of text that we wish to print out\n"\
      "but according to Python style guides, we shouldn't write lines of code\n"\
      "greater than 79 characters long. Hence using the continuation character,\n"\
      " I can get around this restriction.")

The output is now much neater. Let's head back to the idea I had of trying to add text. Let's see what happens in the code sample below.

In [ ]:
print("5" + "5")

This Cleary isn't the output we may have expected - or is it? We can't add text in a numerical sense, as that has no meaning. But we can join text together. This is called concatenation. This is what we have done here, concatenate two text sequences.

##3. Code Blocks
In Python, blocks of code are separated using indentation. Indentation can be expressed using whitespace or tabs. The PEP8 style standard four Python (discussed further down) recommends four spaces per indentation level - this is equal to one-tab character. What does this look like in practice.... take a look at the code in the cell below. Try to understand it, then run it when ready.

In [ ]:
print("This block is not indented.") # First print command.

a = 1
b = 2

if a == 1:
  print("This block is indented by one.") # Second print command.

  if b ==2:
    print("This block is indented by two.") # Third print command.

It may not be obvious what’s happening here - but let's read the code as though it were written in English. it effectively says...


```
Print out the first piece of text.

Set a equal to one.
Set b equal to 2.

if a is equal to one, then print out the text in the block directly below the if.

if a is equal to one and if b is equal to 2, then print out the text in the block directly below the second if.
```


Understanding the **if**-statements doesn't matter for now, we'll come back to them. What matters is that you can see and understand that the code below the if statements is indented. Those indented blocks of code belong together and are independent of one another.


You can think about it this way - the first ```print()``` command is in a block that contains itself, the definitions of $a$ and $b$, and the first **if** statement.

The second print() command is in a block that contains itself and the second **if** statement.

The final ```print()``` command is in a block by itself.

Take a look a the code below and try to determine what block you think the new print statement I've added is in.



In [ ]:
print("This block is not indented.") # First print command.

a = 1
b = 2

if a == 1:
  print("This block is indented by one.") # Second print command.

  if b ==2:
    print("This block is indented by two.") # Third print command.

  print("Which block is this statement in?")

This new piece of code is in the same block as the second print command. This is because they are at the same indentation level.

What we have here is actually a primitive definition of a block of code. But it was necessary to start in this way to build a base to work from. We'll soon learn about new ways to define blocks of code in Python as we pick up some language keywords. These are words with a special meaning that the interpreter understands.

## Code Standards

When writing code, we try to adhere to some common-sense standards. These vary from language to language, based on the syntax. Python has been designed for code simplicity and readability. We've already covered the Zen of Python during the lectures that feeds into this.

<br>

The designer of Python advocated an approach to writing code that was never actually written down. Eventually, style guides and best practices were formalised and written down. Such guides were ultimately incorporated as into Python as the de-facto and recommended approach to Python programming. This was done via a Python Enhancement Proposal (PEP) - a mechanism via which engineers could recommend language improvements. PEP8 included a set of coding style standards. There now exists automated tools that you can use to cheek that your code adheres to such standards. Below is a link to one such tool. I recommend getting into the habit of following the [PEP8 style guide](https://www.python.org/dev/peps/pep-0008/).

[PEP8 Code Standards Checker](http://pep8online.com/)



Let's try some more complicated activities. Each activity below is described and a potential solution that applies (for me at least) is given in some cases. It's up to you to write solutions that work for you.

### Activity 2 - Giant Letters

The goal here is to print out a short version of your name using only the asterisk character. For example, my full name is Robert, but I want to print out Rob so that it looks like the following:

```
****   ****  *****         
*   *  *  *  *   *
*   *  *  *  ****  
* *    *  *  *   *     
*   *  *  *  *   *
*    * ****  *****  
```

You can decide exactly how to print it, but you must use the asterisk. No crazy ASCII art needed

In [ ]:
# Your giant letters attempt.
# Represent Letters pattern SEYE
pattern = {
       'E': ['*****', '*    ', '*****', '*    ', '*****'],
       'S': [' ****', '*    ', '**** ', '    *', '**** '],
       'Y': ['*   *', ' * * ', '  *  ', '  *  ', '  *  '],
       }  # Letter Representation

def letter(nameList):
   for i in range(5):  # length of a list in dictionary is 5
       for j,_ in enumerate(nameList):
           print(pattern[nameList[j]][i]+"   ",end = " ")
       print()

name = "SEYE"   # Print my name as output in asterick
letter(list(name))

 ****    *****    *   *    *****    
*        *         * *     *        
****     *****      *      *****    
    *    *          *      *        
****     *****      *      *****    


### Activity 3 - What you say?

Write an application that can read in some text from a user and print it out.

He's a piece of code that will help you - when you want to retrieve input from a user you can write:

```
txt = input("Type something to test this out: ")
```

The user input is stored in ```txt``` but you could call this anything (this is a variable called txt, and we'll talk about variables in detail soon).

In [ ]:
# Your code here
# input Full name at run time and print

first_name = input ("Enter your First Name: ")
middle_name = input ("Enter your middle name: ")
surname = input ("Enter your surname: ")

print("your full name is:", first_name + " " + middle_name + " " + surname)

Enter your First Name: Oluseye
Enter your middle name: Emmanuel
Enter your surname: Fasuan
your full name is: Oluseye Emmanuel Fasuan


### Activity 4 - Hi Sir Isaac (waves).

Newton's second law of motion can be described by theequation:


$resultant force = mass × acceleration$


where

* force (F) is measured in newtons (N).
* mass (m) is measured in kilograms (kg).
* acceleration (a) is measured in metres per second squared (m/s2)

Can your calculate using Python the force required to accelerate me (Pradeep) at 4 metres per second squared? Assume I weight 95 KG.

Calculate the force and print it out.

In [ ]:
# Your code here

# Calculate the resultanct force - mass x acceleration
mass = int (input ("Enter the mass value in kg: "))
acceleration = int (input ("Enter acceleartion value in m/s2: "))
force = (mass*acceleration)
print("The total force required to accelrate pradeep is: ",force)

Enter the mass value in kg: 95
Enter acceleartion value in m/s2: 4
The total force required to accelrate pradeep is:  380


## 4. Variables

Next week we're going to cover what we call **variables**. We'll get a head start here to prepare for that dicussion.

Variables can be thought of as storage boxes kept in computer memory. The Python interpreter takes care of exactly where in memory that is. We use variables to store all sorts of useful information in our programs.


The Python interpreter figures out what type variables are for us. It is up to us as programmers to ensure we don’t mix the types, in ways that aren’t intended.


Here's a summary of the variables types to be found in Python.



| Type   | Examples                      | Python Code Example |
|--------|-------------------------------|---------------------|
|Integer |… -2, -1, 0, 1, 2 …            | x = 1               |
|Float   |… -2.1, -1.4, 0.01, 1.2, 2.8...| x = 0.5675          |
|String  |hello                          | x = “hello”         |
|Boolean |TRUE or FALSE                  | x = True            |


Below is a video which discusses variables in more detail. Please watch this before you continue. If you can see the video, then run the cell first!

In [ ]:
# Import the HTML library
from IPython.display import HTML

# Load the video frm youtube.
HTML('<iframe width="560" height="315" \
src="https://www.youtube.com/embed/cQT33yu9pY8" \
frameborder="0" allow="accelerometer; autoplay; "\
"encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>')

Now we've watched the video, let’s explore variables in more detail - and tackle some challenges. Read through the cell below and execute it when ready - remember to follow the comments for guidance too.

In [ ]:
# Let's create some variables, and play with the "standard library" functions
# that come bult-in to Python that are available to us.
# The code below creates an integer variable. Remember, integers are
# whole numbers (e.g. -1, 0, 1 etc.).
number_1 = 1

# We can verify the data type of the variable using the type() command:
print("Type of variable 'number_1': ", type(number_1))

# We can also obtain unique identifier for the variable too. To do this we use
# the id() command. When a variable is created, it is assigned a unique
# identifier by the Python interpreter. This allows the variable to be
# tracked and managed appropriately in memory. The ID is a bit like a national
# insurance number. Better yet, think of the ID as a pointer to a variable that
# is stored in memory somewhere.
print("ID of variable 'number_1': ", id(number_1))


# Now let's try and change the value of number_1
number_1 = 2

# What impact does this have on the variable ID?
print("ID of variable 'number_1' after updating: ", id(number_1))

We can see above that changing the value of the variable number_1 from 1 to 2, changed it's identifier. So, what happened here? Well, changing the variable created a new variable. This new variable is stored in a different location in memory, which the updated ID points to. What happens if we try and change the value back?

In [ ]:
# Now let's try and change the value of number_1 back to 1
number_1 = 1

# What impact does this have on the variable ID? Compare the value of
# the ID to the value first reported above.
print("ID of variable 'number_1' after changing back: ", id(number_1))

We find that the ID has returned to the original value outputted. This shows that the original value of number_1 is still in memory, and now we're simply pointing back to it. This becomes important in situations where we need to write efficient programs that use as little memory as possible - for example, when writing applications for mobile devices. So keep in mind the variables you declare and how much memory you are using. Only use the variables you actually need. Now we explore some other variables.

In [ ]:
# The code below creates a floating-point variable (a float). Floats
# are numbers with fractional components (e.g. -1.21, 0.001, 1.483 etc.).
number_1 = 1.999

# We can verify the data type of the variable using the type() command:
print("Type of variable 'number_1': ", type(number_1))

# The code below creates a string variable (a str). Remember, strings
# are just textual variables.
text = "a string"

# We can verify the data type of the variable using the type() command:
print("Type of variable 'text': ", type(text))

# Strings support other functions, which can tell us something about their
# characteristics. One such function is the len() command. It returns the
# length of the string.
print("Length of variable 'text': ", len(text))

We can also change the type of our variables by assigning them new values. Consider the code example below. Try to predict what the output will be. Only run it after making a prediction.

In [ ]:
x = 100
print("Type of variable 'x': ", type(x))
print("The value of 'x': ", x)

x = "Pradeep"
print("Type of variable 'x': ", type(x))
print("The value of 'x': ", x)

What happened here? First, we created  x  and assigned it a numerical value. The Python interpreter correctly determined that the numerical type of this value was an integer (a whole number). The interpreter then set aside the correct amount of memory needed to store an integer. When we then updated the value of  x , we assigned it a string – i.e. a textual value. The interpreter realised this and updated the type of  x  for us. Pretty neat.

##Activity 5
Write a program to input Mark 1, Mark 2, and Mark 3 and print the toal and average marks

In [ ]:
# Write your code here

# Declear mark 1,2 and 3 as requested
# find the average of the 3 marks
# round up to 2 decimal figure

mark1 = int (input ("Enter the first mark value: "))
mark2 = int (input ("Enter the second mark value: "))
mark3 = int (input ("Enter the 3rd mark value: "))
total = mark1 + mark2 + mark2
avg = total / 3
print(" The Total mark is: ", total)
print("The average mark is: ", format(avg,(" .2f")))

Enter the first mark value: 76
Enter the second mark value: 56
Enter the 3rd mark value: 46
 The Total mark is:  188
The average mark is:   62.67


### Activity 6 - Home farm.

A farmer wants us to calculate the number of legs of all the animals on his farm. He's a strange chap, but he offers you wads of cash and a year’s supply of free-range hens’ eggs as a reward. Clearly you can't say no, so you decide to help him. You ask him to inventory the type of animals on his farm and he returns this list:

* Sheep for wool.
* Chickens for eggs.
* Alpacas for wool.
* Sturgeon for caviar.

You then ask him how many of each animal he has. He replies cryptically...

* I have a herd of 1,000 sheep.
* I have half as many Alpacas as I have Chickens
* I have as many as chickens as I have sheep divided by $\pi$
* I have a number of sturgeon equal to the number of chickens I have, divided by the number of alpacas minus 1.

You ask him for some absolute numbers, but he assures you those are the best numbers he has *sigh*. Yu realise he's likely deranged, but free eggs is eggs. So, you carry on regardless.

Goals:

1. Calculate the number of animal legs on the farm, assuming all animals are healthy where alpacas have four legs, sheep have four legs, and chickens have two legs.  In case it needs to be said, sturgeon have no legs.
2. Since your nice you decided to work out for the farmer exactly how many of each animal he has.

Use Python to figure this out using variables and print out these values. Write your code in the cell below.


In [2]:
# Write your code here...
# Determine the number of sheep, chicken, alpacas and sturgeon
# pie is represented by 3.142
# print the number of each animals and total number of legs

sheep = 1000
chicken = int (sheep/ 3.142)
alpacas = int (chicken * 0.5)
sturgeon = int ((chicken /alpacas) - 1)
print("The number of farm sheep is: ", format (sheep))
print("The number of farm chicken is: ", format (chicken))
print("The number of alpacas in the farm is: ", format (alpacas))
print("The number of farm sturgeon is: ", sturgeon)
animal_leg = ((sheep * 4) + (chicken * 2 ) + (alpacas * 4 ))
print("Total animal legs in the farm is:  ", animal_leg)

The number of farm sheep is:  1000
The number of farm chicken is:  318
The number of alpacas in the farm is:  159
The number of farm sturgeon is:  1
Total animal legs in the farm is:   5272


### Activity 7 Weight on Pluto

During the Covid-19 lockdown, I've been stuck behind my desk quite a bit. This means, sadly, I've put on some weight. Whilst at the start of the Covid I was trim 83Kg, I'm now pushing 95 Kg. To make myself feel better, I like to think about how my weight is entirely dependent on the mass of the planet on which I live. Thus, while I'm 95Kg on Earth, I reckon I'm pretty in shape on Pluto. The goal here is to calculate my weight on Pluto to make me feel a whole lot better.

Here's the key information you'll need.

* The surface gravity on Pluto pulls objects toward its surface at a rate of 0.063 $g$.
* My weight on pluto is given by the equation: $Weight = Mass \times \textrm{Surface Gravity}$


In [ ]:
# Write your code here.

mass = 95
surface_gravity = 0.063
weight = mass * surface_gravity

print("The Surface gravity weight on pluto: ", weight, " Kg")

The Surface gravity weight on pluto:  5.985  Kg


##Activity 6 and 7 solutions
Please don't look at these (below) until you've attempted the tasks.

<br>
..
<br>
..
<br>
Don't cheat until you're done...
<br>
..
<br>
..
<br>
..
<br>
..
<br>
..
<br>
..
<br>
Sure you're done?
<br>
..
<br>
..
<br>
..
<br>
..
<br>
..
<br>
..
<br>
..
<br>
..
<br>
..
<br>
..
<br>
..
<br>
..
<br>
..
<br>
..
<br>

###Activity 6 Solution
The crazy farmer has:

1,000 sheep.
Approximately 318 chickens.
Approximately 159 alpacas.
Exactly 1 sturgeon.
The total number of legs on the farm is 5273.2406201955055. Clearly the farmer must be crazy, as he seems to have partial animals - or his math was off. Here's one way to solve this....

In [ ]:
sheep = 1000
chickens = sheep / 3.14159 # a rounded versin of pi.
alpacas = chickens / 2
sturgeon = (chickens / alpacas) -1

print("The farmer has ", sheep, " sheep.")
print("The farmer has ", chickens, " chickens.")
print("The farmer has ", alpacas, " alpacas.")
print("The farmer has ", sturgeon, " sturgeon.")

print("Total animals: ", sheep + chickens + alpacas + sturgeon)

legs = (sheep * 4) + (chickens * 2) + (alpacas * 4)

print("Total legs: ", legs)

### Activity 7 Solution
My weight on pluto is, thankfully, 5.985 Kg.

In [5]:
# Calculate your weight on pluto
# Weight is mass x surface gravity
mass = 95
surface_gravity = 0.063
weight = mass * surface_gravity

print(" Seye's current weight is : ", weight,"Kg" )

 Seye's current weight is :  5.985 Kg
